foundry/glm/glm.py

```diff
@@ -681,30 +703,34 @@ def estimate_laplace_coefs(self, X: ModelMatrix, y: ModelMatrix, sample_weight:
 
         # create mvnorm for laplace approx:
         with torch.no_grad():
-            fail_msg = ''
+
+            self.converged_ = False
+
             try:
-                cov = torch.inverse(hess)
-            except RuntimeError as e:
-                fail_msg = str(e)
+                # cov = torch.inverse(hess)  # Hello floating point errors my old friend.
+                cholesky_hess = torch.linalg.cholesky(hess)  # This is way more numerically stable
+                inv_cholesky_hess = torch.linalg.inv(cholesky_hess)
+                cov = inv_cholesky_hess.T @ inv_cholesky_hess
```

Comment:
just to clarify: you've confirmed that

```python
chol = torch.linalg.inv(torch.linalg.cholesky(hess))
cov = chol.T @ chol
```

 is more stable than
 
 ```python
 cov = torch.linalg.inv(hess)
 ```
 
 ? 
 
 I wonder why that is, since they both require inversion.
 
 If this is right, I think you can skip calculating `cov` and just pass `torch.distributions.MultivariateNormal(scale_tril=inv_cholesky_hess)`.